### PR TITLE
(PA-1942) Update OpenSSL to 1.1.0 in puppet-agent#master

### PR DESCRIPTION
**Pull request by McdonaldSeanp**

### CODE REVIEW

**Comment by McdonaldSeanp** [comment]
jenkins please test this with puppet-agent#master on aix61-POWERa,solaris10-64a,solaris10-SPARCa,solaris11-SPARCa,windows2012r2-64a

**Comment by McdonaldSeanp** [comment]
jenkins please test this with puppet-agent#master on ,aix61-POWERfa,amazon6-64a,amazon7-64a,arista4-32a,ciscoxr-64a,cumulus25-64a,debian8-64a,debian8-32a,debian9-64a,debian9-32a,fedora26-64a,fedora27-64a,osx1012-64a,osx1013-64a,redhat5-64a,redhat6-64a,redhat6-32a,redhat7-64a,redhat7-POWERa,redhat7-AARCH64a,redhatfips7-64a,sles11-64a,sles11-32a,sles12-64a,sles12-POWERa,solaris10-64a,solaris10-SPARCa,solaris11-64a,solaris11-SPARCa,ubuntu1404-64a,ubuntu1404-32a,ubuntu1604-64a,ubuntu1604-32a,ubuntu1604-POWERa,ubuntu1804-64a,windows10ent-32a,windows2012r2_ja-64a%7Blocale=ja%7D

**Comment by McdonaldSeanp** [comment]
jenkins please test this with puppet-agent#master on aix61-POWERfa,amazon6-64a,amazon7-64a,arista4-32a,ciscoxr-64a,cumulus25-64a,debian8-64a,debian8-32a,debian9-64a,debian9-32a,fedora26-64a,fedora27-64a,osx1012-64a,osx1013-64a,redhat5-64a,redhat6-64a,redhat6-32a,redhat7-64a,redhat7-POWERa,redhat7-AARCH64a,redhatfips7-64a,sles11-64a,sles11-32a,sles12-64a,sles12-POWERa,solaris10-64a,solaris10-SPARCa,solaris11-64a,solaris11-SPARCa,ubuntu1404-64a,ubuntu1404-32a,ubuntu1604-64a,ubuntu1604-32a,ubuntu1604-POWERa,ubuntu1804-64a,windows10ent-32a,windows2012r2_ja-64a%7Blocale=ja%7D

File: configs/projects/_shared-agent-settings.rb [hunk 1]
```diff
@@ -196,7 +195,7 @@
   'no-dtls1',
   'no-idea',
   'no-seed',
-  'no-ssl2-method',
+  # 'no-ssl2-method',
```
Comment (caseywilliams) [inline configs/projects/_shared-agent-settings.rb]: Is this option deprecated in 1.1.0?

Comment (McdonaldSeanp) [reply configs/projects/_shared-agent-settings.rb]: yeah, I forgot to move this in to ssl version specific settings

**Comment by McdonaldSeanp** [comment]
I think we should make the version selection pretty deliberate, so I made additions to the other projects to pick openssl 1.0.2,

Those projects will have the option to move to 1.1.0 if they so choose, for now this change shouldn't also effect them

File: configs/projects/_shared-agent-settings.rb (hunk 1 above)
Comment (McdonaldSeanp) [reply configs/projects/_shared-agent-settings.rb]: Updated.

**Comment by McdonaldSeanp** [comment]
superseded by https://github.com/puppetlabs/puppet-runtime/pull/82

**Pull request closed without merging**